src/Datadog.Trace/Util/IdGenerator.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+
+namespace Datadog.Trace.Util
+{
+    internal class IdGenerator
+    {
+#if !NETSTANDARD
+        private static readonly Random GlobalRandom = new Random();
+#endif
+
+        [ThreadStatic]
+        private static Random _threadRandom;
+
+        public static ulong NewSpanId()
+        {
+            var random = GetRandom();
+
+            long high = random.Next(int.MinValue, int.MaxValue);
+            long low = random.Next(int.MinValue, int.MaxValue);
+
+            // Concatenate both values, and truncate the 32 top bits from low
+            var value = high << 32 | (low & 0xFFFFFFFF);
+
+            return (ulong)value & 0x7FFFFFFFFFFFFFFF;
+        }
+
+        private static Random GetRandom()
+        {
+            var random = _threadRandom;
+
+            if (random == null)
+            {
+#if NETSTANDARD
+                random = new Random();
+#else
+                // On .net framework, the clock is used to seed the new random instances
+                // Resolution is too low, so if a bunch of threads are created at the same time,
```

Comment:
Perhaps link to https://docs.microsoft.com/en-us/dotnet/api/system.random.-ctor?view=netcore-3.1 in the comment. It describes the issue and how this is fixed in Net Core

Reply:
Alright